File: testng-core/src/test/java/test/hook/samples/ConfigurableSuccessSample.java
package test.hook.samples;

import static test.hook.HookableTest.HOOK_INVOKED_ATTRIBUTE;
import static test.hook.HookableTest.HOOK_METHOD_PARAMS_ATTRIBUTE;

import org.testng.IConfigureCallBack;
import org.testng.ITestResult;
import org.testng.annotations.Test;

public class ConfigurableSuccessSample extends BaseConfigurableSample {
  @Override
  public void run(IConfigureCallBack callBack, ITestResult testResult) {
    testResult.setAttribute(HOOK_INVOKED_ATTRIBUTE, "true");
    testResult.setAttribute(HOOK_METHOD_PARAMS_ATTRIBUTE, callBack.getParameters());
    callBack.runConfigurationMethod(testResult);
  }

  @Test
  public void hookWasRun() {}
}

File: testng-core/src/test/kotlin/test/hook/samples/ConfigurableSuccessSample.kt
package test.hook.samples

import test.hook.HOOK_INVOKED_ATTRIBUTE
import test.hook.HOOK_METHOD_PARAMS_ATTRIBUTE

import org.testng.IConfigureCallBack
import org.testng.ITestResult
import org.testng.annotations.Test

class ConfigurableSuccessSample : BaseConfigurableSample() {

    override fun run(callBack: IConfigureCallBack, testResult: ITestResult) {
        testResult.setAttribute(HOOK_INVOKED_ATTRIBUTE, "true")
        testResult.setAttribute(HOOK_METHOD_PARAMS_ATTRIBUTE, callBack.parameters)
        callBack.runConfigurationMethod(testResult)
    }

    @Test
    fun hookWasRun() {
    }
}
